### ISSUE TITLE
[Bug] Crash while playing dungeon

### ISSUE THREAD

**Issue by NathanaelGT** [BiscuitDevelopment/SkyblockAddons#449]
Please note that we do not provide support for clients other than Forge.
We will not fix compatibility issues with other clients.

**Describe the bug**
Crash while playing dungeon (happened twice in few weeks)

**To Reproduce**
Steps to reproduce the behavior (how did you get this bug to occur):
1. Play dungeon

**Crash Report**
```
java.lang.IllegalArgumentException: Color parameter outside of expected range: Red
	at java.awt.Color.testColorValueRange(Color.java:310)
	at java.awt.Color.<init>(Color.java:395)
	at java.awt.Color.<init>(Color.java:369)
	at java.awt.Color.<init>(Color.java:461)
	at codes.biscuit.skyblockaddons.listeners.RenderListener.drawText(RenderListener.java:1151)
	at codes.biscuit.skyblockaddons.core.Feature.draw(Feature.java:321)
	at codes.biscuit.skyblockaddons.listeners.RenderListener.renderOverlays(RenderListener.java:344)
	at codes.biscuit.skyblockaddons.listeners.RenderListener.onRenderRegular(RenderListener.java:141)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_146_RenderListener_onRenderRegular_Post.invoke(.dynamic)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:49)
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:140)
	at Apec.Components.Gui.GuiIngame.ApecGuiIngameForge.post(ApecGuiIngameForge.java:399)
	at Apec.Components.Gui.GuiIngame.ApecGuiIngameForge.func_175180_a(ApecGuiIngameForge.java:167)
	at net.minecraft.client.renderer.EntityRenderer.func_181560_a(EntityRenderer.java:1396)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1051)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:349)
	at net.minecraft.client.main.Main.main(SourceFile:124)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
```

**System Information**
 - OS: Windows 10
 - Minecraft Version: 1.8.9
 - Forge Version: 11.15.1.1855
 - SkyblockAddons Version: 1.6.0

**Other Installed Mods**
If you have any other mods installed, please list them here along with their version number.
- Apec 1.7
- AutoGG 4.0.3
- AutoTip 3.0.1
- ctjs 1.3.0
  - Dungeon utilities
  - SkyblockCommands
  - soopyAddons
- Optifine HD U L5
- Orange's Simple Mods 1.2
- Patcher 1.4
- ReAuth 3.5.0
- Scrollable Tooltips 1.3
- Skypixel 0.0.6

**Comment by iHDeveloper** [comment]
Which beta are you using? @NathanaelGT

**Comment by NathanaelGT** [comment]
> Which beta are you using? @NathanaelGT

1.6.0-beta.6

**Comment by ffoxxeyy** [comment]
> > Which beta are you using? @NathanaelGT
> 
> 1.6.0-beta.6

It's the color of something you set that only appears in dungeons like (map,death counter)

**Comment by TimoLob** [comment]
The same bug happens to me in the room which can go up to 3/1 secrets
![2020-11-10_18 28 16](https://user-images.githubusercontent.com/31860160/98712102-25b2d980-2386-11eb-956a-7d5fb6531a6a.png)

The bug seems to happen in RenderListener.java in drawText if Secret Display is enabled.
A percentage is calculated (percent=secrets/maxSecrets). Since secrets=3 and maxSecrets=1 percent becomes 3.
This results in a negative color value which crashes the game.

**Comment by iHDeveloper** [comment]
Thank you @TimoLob for the information about the bug.
I never knew that there's a room with found secrets bigger than max. 😅 
But, I provided a patch based on your patch with some improvements.

Can you test the patch and check if it works with the bugged dungeon room? 🤔 
```sh
git remote add ihdeveloper https://github.com/iHDeveloper/SkyblockAddons
git checkout ihdeveloper/dungeon/fix-secrets-crash
```

**Comment by Phoube** [comment]
I think everything here should be addressed in 1.6.0. Appreciate the bug report, and sorry it took so long for us to close. Feel free to re-open if any problems persist!
